### ISSUE TITLE
sqlfluff github org

### ISSUE THREAD

**Issue by alanmcruickshank** [sqlfluff/sqlfluff#444]
@NiallRees had an excellent suggestion. Rather than sqlfluff just living on my private github account, maybe the time has come for it to have it's own github org (https://github.com/sqlfluff)

This is a callout for two things:
1. Feedback on how having `sqlfluff/sqlfluff` would impact people's contributions to sqlfluff (rather than `alanmcruickshank/sqlfluff`, as it is now).
2. If we do migrate the repository, what things should we do at the same time? e.g. Should it be part of a major release? Can it happen any time? In particular if anyone has any learnings from similar migrations.
3. Would any of the more active committers want to be part of that organisation and consider becoming part of the team that can review PRs and other community activities? I'm thinking particularly of @barrywhart , @pwildenhain, @dmateusp, @nolanbconaway & @flpezet, but I'm curious for any other volunteers.

**Comment by dmateusp** [comment]
1. I think creating the org would allow org members to share responsibilities and ownership beyond just committing code, it would be nice to have more "meta" discussions around roadmap, priorities, brainstorm solutions etc

2. No experience on that! I'm reading the docs [here](https://help.github.jp/enterprise/2.11/user/articles/transferring-a-repository-owned-by-your-personal-account/), not sure about doing a major release at the same time, I think compatibility with DBT packages would be a good candidate for a `1.0`

3. Totally!

**Comment by barrywhart** [comment]
I'd be happy to participate in the organization! I agree moving the repo could encourage more contributions.

About point 2. I'm leaning against a major release, but another thought... do any folks here blog, either independently or for your employer? My employer, Mailchimp, recently started a tech blog. It could be cool to create some publicity for the project by blogging or otherwise publicizing how you are using SQLFluff and to tie it to any public marketing for projects it helped with , e.g. this [Mailchimp page](https://mailchimp.com/smart-marketing/).

**Comment by nolanbconaway** [comment]
Yeah this makes sense to me. I don't think anyone has broached the issue of text editor integrations yet, but these would almost certainly require additional repos. Creating an org would be a great way to collect these repos together.

**Comment by alanmcruickshank** [comment]
> do any folks here blog, either independently or for your employer? 

I've got both a personal blog (which has a few sqlfluff articles, but nothing very meaty http://designingoverload.com/tag/sqlfluff/ ) and an employer blog ( https://lifeat.tails.com/author/alanmcruickshank/ ) but I've not posted anything explicitly on the latter about sqlfluff (which I probably should). I'd love some additional content out there about how people use it though! Very happy to assist anyone wanting to write a blog about it.

**Comment by flpezet** [comment]
Hello ! I feel honored to be potentialy included in the team! 
For now, I prefer to not join the team. I don't think I will be able to give somewhat consistent contributions in the coming months. 
But thank you for thinking of me! :)

**Comment by pwildenhain** [comment]
I think moving to an org is a great idea, and I'm in to be on the team 💪 

@nolanbconaway:
> I don't think anyone has broached the issue of text editor integrations yet

Check this out: https://github.com/dorzey/vscode-sqlfluff I've started using it and it works like a charm 🥇 (though it is only for VSCode)

@alanmcruickshank :
> Should it be part of a major release?

I would be interested to hear what milestone's you would like to achieve before releasing 0.4.0. I see your milestone here: https://github.com/alanmcruickshank/sqlfluff/milestone/3 -- are you still interested in tackling those?

**Comment by nolanbconaway** [comment]
> Check this out: https://github.com/dorzey/vscode-sqlfluff I've started using it and it works like a charm 🥇 (though it is only for VSCode)

_Excellent, just what i was looking for!~

**Comment by alanmcruickshank** [comment]
So.... it's done!

We now have:
- https://github.com/sqlfluff
- https://github.com/sqlfluff/sqlfluff

You can still go to https://github.com/alanmcruickshank/sqlfluff, and github will safely redirect. I've checked issue assignments and it looks like everything has survived the transition. Can the people on this thread do a quick sanity check that everything seems to still work on their side?

I've also added the people who volunteered on this thread to the organisation. 👍 

I totally hear the questions around what we need to crack before the next major milestone. I'd like to firm that up in due course. I think a good way of doing that might be the "projects" feature of the new org. Unless anyone has any other good ideas for doing this?

**Comment by nolanbconaway** [comment]
@alanmcruickshank , I'd be happy to move [sqlfluff-online](https://github.com/nolanbconaway/sqlfluff-online) to the new org if desired! Also pls add me if it is low cost (my contribution will be low volume for a good but but i do love this project <3)

**Comment by pwildenhain** [comment]
I've found that milestones are great for tracking what will go into a specific _release_ while projects are good for prioritizing/showing the status of issues. I use both at my work, and I would recommend the same 👍

**Comment by alanmcruickshank** [comment]
> @alanmcruickshank , I'd be happy to move sqlfluff-online to the new org if desired! Also pls add me if it is low cost (my contribution will be low volume for a while but i do love this project <3)

If you're happy to then that would be awesome - I've sent you an org invite. Would you mind either adding a "contributing" section or a "contributing.md" file in the repo so that it's clear how you want people to interact with the repo going forward? Similar to https://opensource.creativecommons.org/contributing-code/github-repo-guidelines/.

> and to @nolanbconaway's point about adding more repos to the org, I can reach out to the owner of the VSCode plugin to see if he's interested in transitioning his repo here.

If you're happy to, that would be great. I think that would help people discover his awesome project much more easily. I don't want it to feel like his project is being stolen at all, and I'd still love him to continue to develop and support it. If they're keen let me know here and I'll drop them an invite to the org. I think they'll maintain *most* of the same controls, there are two tiers of user within an org, _members_ and _owners_. So far I've only added people as _members_, but I think the only thing that restricts is destructive stuff like permanently deleting repos and transfers out. Especially with public repos, they basically work exactly the same as otherwise. If they've got any concerns, then feel free to funnel them here - or if it doesn't work out I'm also happy to migrate it back out of the org if it's too much of a pain. Totally up to them.

I've also added the sourcecode for sqlfluff.com [https://github.com/sqlfluff/sqlfluff.com] to the org too as a private repository. I don't think it's where I want to direct people who want to contribute straight away, but for people who help maintain the projects we're collecting together - I think it makes sense to collaborate on that too [although be nice about my really shoddy javascript!! 😬 ].

I'll have a go at updating some potential roadmap items later this week so we can talk properly about 0.4.0.

**Comment by alanmcruickshank** [comment]
Project for 0.4.0 is live: https://github.com/sqlfluff/sqlfluff/projects/2

There's a column on the far left called "shortlist". If people have got particular issues or features they think should be in the next major release can they drop them in there. Let's see what floats to the top and then we can hash out what makes it through to the next phase.

Aim for this phase is to get some rough agreement on what we want the next big goals to be.

**Comment by nolanbconaway** [comment]
@alanmcruickshank contributing is now there for sqlfluff-online, and i moved it over. Await access your 👍  for heroku deploys, but i also had some configuring to do so can i also get some admin privs to that repo?

**Comment by pwildenhain** [comment]
@alanmcruickshank I don't think I have the permissions to add cards

![image](https://user-images.githubusercontent.com/35195136/94757690-ff525500-0368-11eb-8c2e-053029b79403.png)

Or to tag issues

![image](https://user-images.githubusercontent.com/35195136/94757711-1002cb00-0369-11eb-9177-63599fffbfe3.png)

**Comment by alanmcruickshank** [comment]
I've updated permissions.

- I've created a `sqlfluff-maintainers` team, with everyone that a previously added to the org.
- On sqlfluff/sqlfluff, the maintainers have *triage* access which means people can managing issues and PRs. Eventually I'd like to up this to write and maintain access, but I wonder whether it's worth doing this one step at a time for now.
- On sqlfluff/sqlfluff-online, @nolanbconaway you should have full admin privileges, and I've made the maintainers team also have triage access there so the same group can manage issues and PRs there too.

@pwildenhain can you see whether that means you can now edit the project?

**Comment by nolanbconaway** [comment]
> * On sqlfluff/sqlfluff-online, @nolanbconaway you should have full admin privileges, and I've made the maintainers team also have triage access there so the same group can manage issues and PRs there too. Can you check that does everything you need it to?

Did you get this notify @alanmcruickshank ? This is what we're waiting on.

![image](https://user-images.githubusercontent.com/1770896/94811795-db702d00-03c3-11eb-8ea7-111cf91a8c86.png)

Besides that, we might consider making a sqlfluff heroku account (as right now online is deployed via my personal account).

**Comment by alanmcruickshank** [comment]
Thanks @nolanbconaway : Is that sorted now? I think I've clicked the right button. As for a sqlfluff heroku account, I'm open to that - it could be a good idea. If we do go down that road, I'd love to have as much of the infrastructure setup in configuration rather than manually set up so that we could migrate it if we wanted. I've got the sqlfluff website on AWS right now, and I probably need to do the same for that too. My favourite tool for such things is terraform, which looks like it also has heroku bindings (although I've never used them personally): https://registry.terraform.io/providers/heroku/heroku/latest/docs

Thanks to @NiallRees for checking on board access. People *couldn't* edit the board, I've now upped everyone to `write` access so now they can. Go nuts!

**Comment by pwildenhain** [comment]
🎉 Yay for project management (it shouldn't be this fun to add labels to things)

Right off the bat, I realize that I might be  deviating from the original purpose of this issue. I definitely don't want to spam people, so please speak up 🗣️ if you think this discussion should happen elsewhere 🚚 

I combed through the backlog today, trying to familiarize myself with the current state of the project, and organize for the upcoming release. Here's what I did:

### Add issues to the project shortlist

My criteria were:
- Issues that we in the old `0.4.0` milestone
- Bugs that seem like they have a fix in mind
- Features that I'm personally interested in

### Add new issue labels

Similar to how there were `parsing` and `new rule` labels, I wanted to add labels that reflected other parts of the package infrastructure.

I added:
- 🔗 [dialect](https://github.com/sqlfluff/sqlfluff/issues?q=is%3Aopen+is%3Aissue+label%3Adialect+). For issues related to specific SQL dialects. I toyed with the idea of a label for each dialect, but that seemed like overkill.
- 🔗 [templating](https://github.com/sqlfluff/sqlfluff/issues?q=is%3Aopen+is%3Aissue+label%3Atemplating). Mostly issues about `dbt` macros


### Thought about the 0.4.0 release

I think the major release should ideally squash as many bugs 🐛 👢 as possible. Which of these do y'all think we should tackle? 
🔗 [List of unassigned bugs](https://github.com/sqlfluff/sqlfluff/issues?q=is%3Aopen+is%3Aissue+label%3Abug+no%3Aassignee)

The other common thread that kept popping up was issues with how sqlfluff interacts with a`dbt` project. The ideas seemed to be numerous on how to best address this -- by far my favorite was to depend on `dbt` to handle `dbt` things (like macros). The thought being that if the user has templated SQL, they likely already have `dbt` installed for same project. Whatever the solution, I think this would also be a terrific ✨ enhancement for the next a major release.

**Comment by barrywhart** [comment]
We don't use dbt, but we do use templating. We might be in the minority here. What does "depending on dbt" look like? Is it as simple as creating a dbt configuration file, or something heavier? I think we can adapt if it's not too onerous.

**Comment by barrywhart** [comment]
About bug fixes, 366 is an actual crash, so I'd say that's high priority. Ideally, I'd like to address all parsing issues, or if not, in priority order based on how many people will encounter them.

**Comment by pwildenhain** [comment]
> What does "depending on dbt" look like?
This is a really good question. I was thinking that `dbt` would more "enhance" the `sqlfluff` experience. Maybe a good solution look like keeping all the `jinja` functionality we have now, and _adding_ `dbt` specific support for macros and dbt packages. That you don't _need_ `dbt` for `sqlfluff` templating to work **unless**  you want tighter integration with your existing `dbt` project. 🤔 Don't know enough about the project though to suggest a more specific implementation

**Comment by alanmcruickshank** [comment]
> speak up 🗣️ if you think this discussion should happen elsewhere

I think we should probably close this issue and have a place specific for 0.4.0. I've just set that up here: https://github.com/sqlfluff/sqlfluff/issues/470

@pwildenhain , would you mind reposting your excellent summary there? I totally agree that the discussion that we're having about the dbt and macro dependency is the biggie that we need to hash out. In particular whether we should try and support the extra dbt related features using some kind of plugin architecture as mentioned in #164 , or whether we build it into the core project.

**Comment by alanmcruickshank** [comment]
To close out this specific issue, I've put together a brief PR together for how I think the current process is looking: #471.

If we can get broad agreement on the process in that PR, then I'm tempted to close out this issue and we can crack on with the more detailed conversation for 0.4.0 above. Unless there are other project governance things that people think it's worth putting in place at this stage?

**Comment by pwildenhain** [comment]
> 
> 
> We don't use dbt, but we do use templating. We might be in the minority here. What does "depending on dbt" look like? Is it as simple as creating a dbt configuration file, or something heavier? I think we can adapt if it's not too onerous.

**Comment by pwildenhain** [comment]
agree, let's close this with #470 and move on to planning 0.4.0 specifically
